README.md

```diff
@@ -901,6 +901,52 @@ type TotallyMutableFoo = Mutable<Foo>
 
 `Readonly` is a type mapper to make all properties of an object to be readonly.
 
+## Enums
+[Official Documentation](https://www.typescriptlang.org/docs/handbook/enums.html)
+> Enums allow us to define a set of named constants. Using enums can make it easier to document intent, or create a set of distinct cases. TypeScript provides both numeric and string-based enums.
+
+Notice that enums emit extra code, meaning its not only a type annotation feature.
+
+Enums values help with maintainance, you can easily find usages, and rename enum values (compared to string literal types).
+
+Enum support number as well as string values (also mixure though probably not useful).
+
+You can get in runtime, an enum value from its name, and its name from the value:
+```ts
+enum MyEnum { Foo, Bar }
+
+let nameOfFoo = MyEnum[MyEnum.Foo]; // "Foo"
+let valueOfFoo = MyEnum["Foo"]; // 0
+```
+
+Enums allow you to get the keys, value, and iterate over them:
+```ts
+enum MyEnum { Foo, Bar }
+
+let keys = Object.keys(MyEnum); // ["Foo", "Bar"]
```

Comment:
let keys = Object.keys(MyEnum); // ["0", "1", "Foo", "Bar"]

Reply:
Correct. My bad. Thank you!